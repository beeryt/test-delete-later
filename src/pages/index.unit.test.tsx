import React from 'react'
import { render, waitFor } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

import IndexPage from './index'

describe('IndexPage', () => {
  const siteMetadata = {
    title: 'Site Title',
    description: 'Site Description',
    siteUrl: 'example.com'
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return { site: { siteMetadata: siteMetadata } }
    })
  })

  afterAll(() => useStaticQuery.mockReset())

  it('renders correctly', async () => {
    render(<IndexPage />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() => expect(document.title).toBe(siteMetadata.title))
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })
})
