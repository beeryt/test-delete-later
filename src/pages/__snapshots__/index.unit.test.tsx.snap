// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<body>
  <div>
    <header>
      <nav
        aria-label="main navigation"
        class="navbar is-primary"
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item is-size-3 has-text-dark"
            href="/"
          >
            Site Title
          </a>
          <div
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger "
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </div>
        </div>
        <div
          class="navbar-menu "
        >
          <div
            class="navbar-end"
          />
        </div>
      </nav>
    </header>
    <main
      id="full-body-wrapper"
    >
      <section
        class="section"
        id="about-me"
      >
        <div
          class="columns container"
        >
          <div
            class="column is-one-quarter"
          >
            <div
              class="card"
            >
              <div
                class="card-image"
              >
                <figure
                  class="image"
                >
                  <img
                    alt="Placeholder image"
                    src="https://bulma.io/images/placeholders/1280x960.png"
                    title=""
                  />
                </figure>
              </div>
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <p
                      class="title is-4"
                    >
                      Carl B. Smiley
                    </p>
                    <p
                      class="subtitle is-6"
                    >
                      @linux_pangolin
                    </p>
                  </div>
                </div>
                <div
                  class="level is-mobile"
                >
                  <a
                    aria-label="GitHub"
                    class="level-item"
                    href="https://github.com/beeryt"
                  >
                    <span
                      class="icon"
                    >
                      <i
                        class="fab fa-github"
                      />
                    </span>
                  </a>
                  <a
                    aria-label="LinkedIn"
                    class="level-item"
                    href="https://www.linkedin.com/in/carl-b-smiley"
                  >
                    <span
                      class="icon"
                    >
                      <i
                        class="fab fa-linkedin"
                      />
                    </span>
                  </a>
                  <a
                    aria-label="StackExchange"
                    class="level-item"
                    href="https://stackexchange.com/users/13595322/linux-pangolin"
                  >
                    <span
                      class="icon"
                    >
                      <i
                        class="fab fa-stack-overflow"
                      />
                    </span>
                  </a>
                  <a
                    aria-label="Twitter"
                    class="level-item"
                    href="https://twitter.com/linux_pangolin"
                  >
                    <span
                      class="icon"
                    >
                      <i
                        class="fab fa-twitter"
                      />
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="column"
          >
            <article
              class="article content"
            >
              <h1
                class="title"
              >
                About Me
              </h1>
              <p>
                I am an embedded systems engineer with practical experience in both hardware and software.
              </p>
              <p>
                I currently work at
                 
                <a
                  href="https://www.protingent.com"
                >
                  Protingent
                </a>
                 as an embedded systems engineer for
                 
                <a
                  class="nobr"
                  href="https://strykeremergencycare.com"
                >
                  Stryker Emergency Care
                </a>
                .
              </p>
              <p>
                I am located in the Seattle metropolitan area.
              </p>
              <p>
                For professional inquiries, please
                 
                <a
                  href="https://www.linkedin.com/in/carl-b-smiley"
                >
                  visit my LinkedIn
                </a>
                .
              </p>
            </article>
          </div>
        </div>
      </section>
      <section
        class="section"
      >
        <article
          class="article container"
        >
          <h1
            class="title"
          >
            Welcome
          </h1>
          <h2
            class="subtitle"
          >
            For best results, follow directions carefully.
          </h2>
          <div
            class="media"
          />
          <a
            class="media"
            href="#"
          >
            <figure
              class="media-left image is-64x64"
            >
              <img
                alt=""
                src="https://bulma.io/images/placeholders/128x128.png"
                title=""
              />
            </figure>
            <div
              class="media-content"
            >
              <p
                class="title is-4"
              >
                Project 1
              </p>
              <p
                class="subtitle is-6"
              >
                Do not write in this space.
              </p>
            </div>
          </a>
          <a
            class="media"
            href="#"
          >
            <figure
              class="media-left image is-64x64"
            >
              <img
                alt=""
                src="https://bulma.io/images/placeholders/128x128.png"
                title=""
              />
            </figure>
            <div
              class="media-content"
            >
              <p
                class="title is-4"
              >
                Project 2
              </p>
              <p
                class="subtitle is-6"
              >
                Stay seated until bus comes to a complete stop.
              </p>
            </div>
          </a>
        </article>
      </section>
    </main>
    <footer
      class="footer"
    />
  </div>
</body>
`;

exports[`IndexPage renders correctly 2`] = `
<head>
  <title>
    Site Title
  </title>
  <meta
    content="Site Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;
